records_mover/records/job/config.py

```diff
@@ -29,8 +29,13 @@ def non_default_arg_names(self, fn: Callable[..., Any]) -> List[str]:
 
     def add_hint_parameter(self, kwargs: Dict[str, Any], param_name: str) -> None:
```

Comment:
First complication - not every argument in a source/target factory method maps 1:1 to a command line parameter.  The example we're hitting here is a RecordsFormat object - which has two subtypes (parquet and delimited).  The DelimitedRecordsFormat() constructor takes both a variant argument and a hints dict, which can contain various keys.

We have a technique (which will show up further down in this PR) for exploding things like RecordsFormat out into multiple command line arguments.

This function that we're looking at now is on the call path when we're accepting the parsed command line argument results *back in* (as the `kwargs` dictionary) and are formulating the call into the factory method.